ERCS/erc-7824.md

```diff
@@ -0,0 +1,146 @@
+
+## Specification
+
+### Key Structs and Enums
+
+#### `Outcome`
+Defines the result of asset redistribution upon channel closure.
+
+```solidity
+struct Outcome {
+    SingleAssetExit[] singleAssetExits; // Array of asset-specific exits
+}
+```
+
+#### `SingleAssetExit`
+Represents allocations for a single asset.
+
+```solidity
+struct SingleAssetExit {
+    address asset;                  // Asset type (e.g., native token or ERC20)
+    bytes assetMetadata;            // Metadata for non-standard assets
+    Allocation[] allocations;       // Distribution rules for this asset
+}
+```
+
+#### `Allocation`
+Details how an asset is distributed.
+
+```solidity
+struct Allocation {
+    bytes32 destination;            // Address or channel ID receiving the funds
+    uint256 amount;                 // Amount to allocate
+    uint8 allocationType;           // Type of allocation (e.g., simple or guarantee)
+    bytes metadata;                 // Additional data for advanced allocations
+}
+```
+
+### Allocation Types
+
+1. **Simple Allocation (`allocationType = 0`)**  
+   Basic asset transfer to external addresses or channels.
+
+2. **Guarantee Allocation (`allocationType = 1`)**  
+   Conditional transfer based on secondary channels or other criteria.
+
+```solidity
+enum AllocationType {
+    Simple,
+    Guarantee
+}
+```
+
+### Advanced Metadata Example: Guarantees
+
+For `Guarantee` allocations, the `metadata` encodes the following structure:
+
+```solidity
+struct Guarantee {
+    bytes32 left;  // Left participant in the guarantee
+    bytes32 right; // Right participant in the guarantee
+}
+```
+
+### Example Usage
+
+#### Example 1: Simple ETH Allocation
+"5 ETH to Alice, 5 ETH to Bob."
+
+```solidity
+SingleAssetExit ethExit = SingleAssetExit({
+    asset: address(0), // Native token (ETH)
+    assetMetadata: "",
+    allocations: [
+        Allocation({
+            destination: 0x...AliceAddress,
+            amount: 5 ether,
+            allocationType: AllocationType.Simple,
+            metadata: ""
+        }),
+        Allocation({
+            destination: 0x...BobAddress,
+            amount: 5 ether,
+            allocationType: AllocationType.Simple,
+            metadata: ""
+        })
+    ]
+});
+```
+
+#### Example 2: Guaranteed Allocation
+Guarantee that funds can only be moved to a secondary channel.
+
+```solidity
+Allocation({
+    destination: 0x...ChannelID,
+    amount: 10 ether,
+    allocationType: AllocationType.Guarantee,
+    metadata: abi.encode(Guarantee({left: 0x...Alice, right: 0x...Bob}))
+});
+```
+
+## Rationale
+
+The Exit Format standardizes asset redistribution logic across state channel implementations, reducing ambiguity and improving interoperability. By supporting both simple and advanced allocations, it addresses a wide range of use cases, from basic payments to complex virtual channel setups.
```

Comment:
This mostly reads like motivation because it's justifying the proposal as a whole. The rationale should be used to explain why you made certain technical choices within the document. For example, why you've included metadata fields or why `destination` is 32 bytes long. Obviously you should explain actually important stuff 🤣